### DISCUSSION TITLE
How to get access to the interface?

### DISCUSSION THREAD

**Discussion by You69Man** [ehorvat1/NES-MEP-Reader#7] [Q&A]
This is my smart meter:
![grafik](https://github.com/ehorvat1/NES-MEP-Reader/assets/6373072/9ef5e040-d149-44c5-81ce-7746a4c7c636)
I'd really like to implement this project, but obviously the MEP interface of this meter is hidden behind lead seals.

I found this picture from https://www.dabbler.dk/wp-content/uploads/2021/08/08-Peter-K.-Hansen-Develco-Specification-MEP-ESNA.pdf (red correction by my own), that seems to show the inside hidden MEP interface:
![grafik](https://github.com/ehorvat1/NES-MEP-Reader/assets/6373072/b0d3d7bc-8cb3-4d06-b9a6-97d6285df9ce)
![grafik](https://github.com/ehorvat1/NES-MEP-Reader/assets/6373072/34e82241-9510-40dc-bbde-320048862407)

But how to access this? I don't expect the energy provider to allow to break the seals, would they? 
How do you deal with this problem? 

What about the IR interface? Is this also active?
According to the "DATENBLATT SMART-METER-ADAPTER" https://oesterreichsenergie.at/fileadmin/user_upload/Datenblatt_Smart_Meter_Adapter_V7_20220808__2_.pdf (unfortunately still not available to purchase up to now) this doesn't seem to be the case:
![grafik](https://github.com/ehorvat1/NES-MEP-Reader/assets/6373072/fde1205f-bcb1-47aa-9ad3-e699ee765c30)
Anyone who has other qualified information about this?

**Comment by ehorvat1** [comment]
Hi,
please do not break the seals .... you get in deep troubles... please see my answer I gave to Lüder on the other Discussion here: https://github.com/ehorvat1/NES-MEP-Reader/discussions/6 . 
IR was not an option for me.
And also Notes Pt. 1 here: https://github.com/ehorvat1/NES-MEP-Reader/tree/main/Hardware 
I have even still some of those circuits lying around here (Amstetten/NÖ) and would not mind providing you with one. 
Main problem nowadays seems to be the availability of original MAX3232 converters.
BTW, when I have some spare time I want to merge the "ahoy" readout protocoll for Hoymiles inverters also to this web interface.
BR, Bert

**Comment by You69Man** [comment]
Hi Bert,
thanks for your response. Be sure that I wouldn't have broken the seals by my own without asking the supply company ;-)
So i will get in contact with them and ask them to give me access to the interface. Already curious about their answer :)
I'm also a little bit involved into the Ahoy project, so of course it's also my goal to integrate Ahoy with my smart meter data in some way.
Manfred

**Comment by ehorvat1** [comment]
Servus Manfred,
bei mir in Amstetten war das kein Problem da EVU klein genug und der Mann von der Zählerabteilung war selber interessiert.
Falls sich die Netz Linz AG ziert und dir keinen Zugang zur Kundenschnittstelle geben will musst du sie halt fragen ob sie sich nicht an die Gesetze halten wollen.
Bezüglich der Smartmeter gibt es eine öst. Verordnung nämlich die "Intelligente-Messgeräte-AnforderungsVO 2011 – IMA-VO 2011" und dort ist  unter § 3. Intelligente Messgeräte gemäß § 7 Abs. 1 Z 31 ElWOG 2010 im Absätz 6 angeführt:
**Absatz 6. Die intelligenten Messgeräte sind dahingehend auszustatten, dass sie über eine Kommunikationsschnittstelle mit in der Kundenanlage vorhandenen externen Geräten kommunizieren sowie mindestens alle gemäß Z 2 erfassten Daten unidirektional ausgeben können. Der Zugriff sowie die Spezifikationen dieser Schnittstelle sind bei Bedarf allen Berechtigten auf deren Anfrage ab Einbau des intelligenten Messgerätes diskriminierungsfrei zur Verfügung zu stellen. Diese Schnittstelle ist zur unidirektionalen Ausgabe von Daten und Informationen zu konfigurieren. Eine über die oben beschriebene Nutzung hinausgehende Verwendung dieser Schnittstelle zur Steuerung des Zählers ist nicht vorzusehen**

Wobei das Wort "diskriminierungsfrei" wichtig ist, und das würde für mich auch kostenlos bedeuteten....
Bei den Echelon basierenden Smartmetern hat wohl niemand an die bösen Kunden gedacht die ihre Daten selber verwenden wollen. Mich hat ja überrascht dass z.B. die EVN in NÖ das viel besser macht - sind auch andere Zähler. 
Doku der EVN: https://www.netz-noe.at/Download-(1)/Smart-Meter/218_9_SmartMeter_Kundenschnittstelle_lektoriert_14.aspx

LG, Bert
